### DISCUSSION TITLE
Touch Screen Display feature and bug discussion?

### DISCUSSION THREAD

**Discussion by steve8x8** [OpenNeptune3D/OpenNept4une#140] [General]
A while ago I left Discord (couldn't handle the traffic anymore), so I might have missed something important - please bear with me.

Yesterday I finally managed to upgrade my 4pro (that had been running OpenN 0.15 since around Christmas and sitting idle for a while), with a full upgrade including MCU and the display code, and after successfully printing some items that had been in the queue for some time, I looked closer at the display functionality.

I had flashed the 1.2.11 TFT successfully so from the firmware side I should be safe.

I'm also aware that display support is still work in progress, but I couldn't find a list of known bugs (please fill me in...), so here's mine:

While printing, in the "adjust" submenu of "settings" the soft buttons don't work (but the left-arrow on top does), this doesn't affect the "filament" or "speed" submenus.

With the printer now idle,
- "about machine" shows both IPs in one line, so the second one is clipped (what about adding a LF if scrolling or autowrapping, like in "adjust" isn't possible?)
- also in "about machine" clicking on OpenNeptune produces an up-arrow that has no function (this was to upgrade firmware in stock, iirc?)
- in the "extruder" menu, "load" shows no response (cold nozzle?)
- in the "prepare" "move" menu, + and - don't work but homing X or Y adds some Z (even All-Home does this at the beginning)
- in "settings", the "filament detect" toggle is off and cannot be switched on (same in "adjust" while printing)
- in "advanced settings" the "print mode" and "level mode" buttons have no effect
- in "advanced settings" the power loss recovery toggle is off (which is good) but cannot be changed

_Also after having been puzzled by a few rather weird German translations (e.g. "Streckenlampe"): since I'm already on Crowdin, are translations handled there - I'd volunteer to contribute?_

I suppose changes to the display code are handled in the display_connector repo only, so I'll check the "git log" of my clone and https://github.com/OpenNeptune3D/display_connector/commits/dev/  ;)

**Comment by steve8x8** [comment]
At least re: the weird German translations: found them in the TFT file proper, so there's probably no way to get them fixed - I'm fine with English, but some others may not, and such stuff can be demotivating. Good to know though that ON is not to blame.
(BTW, https://de.glosbe.com/de/de/Streckenlampe offers a Chinese translation only, and I'm unable to verify that one. It doesn't explain what "Streckenlampe" might mean.)

**Reply by barrenechea** [reply]
I think this would be better addressed by contacting [Elegoo directly](https://www.elegoo.com/pages/contact-us) since they are developing our screen's firmware.

Hopefully, they'll be able to help you (and all users) with better translations! From my personal experience, they are very responsive.

**Reply by phillipthelen** [reply]
Jeah. With how the display service is written, we _could_ provide patches for weird translations, but… I’m not sure yet I’d want to go down that route. If elegoo ends up being unhelpful, maybe.

**Reply by crazyslicster** [reply]
I installed these firmwares recently and even with the touchscreen having bugs, I think it's much better than the stock Elegoo firmware. I'm wondering why elegoo doesn't simply help to support this and make it their official firmware? They'd have a heck of a lot less development to do on their end.

**Reply by steve8x8** [reply]
@crazyslicster because they would have to support it the moment it bears their name...

**Comment by steve8x8** [comment]
Although there have been no updates in the recent past, I'd like to share my (additional) observations and ideas:

Shutting down the host from the touchpad works (and fans will change speeds a few seconds after clicking "shutdown") but otherwise there's no feedback from the screen - the four big square buttons even change their appearance to "clicked" and play a sound. Since there's no host that could respond to any of those actions, what about clearing the screen (and perhaps showing a switch-off icon or animation)?

For the filament "load" and "unload" actions, I'm missing thermal activity that would be required (170 deg C seem to be the critical point). Some "heat to 170" button would fit between them (and it could be turned into a "heat off" one once the new target temp has been set).

More to come once I've fixed the clogged hotend (TPU is a beast...)

**Comment by S-Kececi** [comment]
I have a problem i cant fix it( I bought second-hand printer (N4 Pro) Last user did some upgrade or smt idk. (It written OpenNept4une) I cant use well the printer and i want to reset/factory settings. But i can't do it. I downloaded last firmware packs but i cant run anyway. Can someone help me?

**Comment by mtmello2003** [comment]
I installed OpenNept4une yesterday, and it seems the same long-standing issues are still unresolved... :(

I use the display a lot and prefer it over the remote options like Fluidd and Mainsail.
